### DISCUSSION TITLE
Error while uploading ipa file with Transporter made with new SDK 33.1.1.345

### DISCUSSION THREAD

**Discussion by danielkotzer** [airsdk/Adobe-Runtime-Support#651] [Q&A] [answered]
## Problem Description

I created an ipa file to upload it to app store, with the new Air SDK 33.1.1.345. got an error message: "Error itms-90023: missing required icon file 167x167". The Assets.car file is fine, made with latest xcode, the xcode is updated, mac updated, everything is fine, when I try to upload with older version of 33.1 sdk, it works, but I wanted to use the latest version of the SDK but this one is not making a valid ipa file that can be uploaded to app store.

* Which AIR SDK did you use? Hint: always try the latest version from [Adobe Labs](http://labs.adobe.com/downloads/air.html).
* 33.1.1.345
* Describe the affected environment: Flash vs. AIR, Android vs. iOS, all devices vs. just specific models.
* flash builder 4.7
* Can you reproduce it yourself, or did you get reports from other users? If so, how many users are affected (percentage)?
* always the same when trying to upload with Transporter app on mac.
* If applicable, link to the Starling/Feathers issues that brought you here.

Furthermore, **always** add a link to one or more analog [Adobe Tracker](https://tracker.adobe.com) issues. This repository is not a replacement for Adobe's bug tracker!

## Steps to Reproduce

Tell us how this issue can be reproduced. Seriously: if you want this issue to be fixed, this is the part you **really** need to get nailed.
Give us a minimal example:

* As little code as possible. Get to the root of the problem!
* As few dependencies as possible. Did the problem pop up with Feathers? Try to reproduce it with Starling only.
* Remove any ANEs (except if they are vital for the problem).
* Include all necessary files (e.g. textures).

## Known Workarounds

Just in case others run into this – do you happen to know of a workaround? Otherwise, just remove this section altogether.

**Comment by marchbold** [comment]
Are you sure you have included this size in your assets.car?

**Comment by marchbold** [comment, marked as answer]
Also if you are supporting older versions of iOS you still need the icon in your app descriptor so check you have:

```xml
<icon>
    <!-- other sizes -->
    <image167x167>icons/icon167x167.png</image167x167>

```

**Reply by danielkotzer** [reply]
Turns out it wasn't a problem in the assets.car, but in the descriptor I had to add these icons, didn't have to do it in the past, everything was fine as long as the icons were in the assets.car, now you have to make sure they are in the descriptor too.

**Comment by marchbold** [comment]
@danielkotzer It will depend on the minimum ios version that you set. If you use 9, then you still need to add the older style ios icons.
